File: src/main/java/legacy/kms/chapter02/discount/PercentDiscountPolicy.java
package legacy.kms.chapter02.discount;

import legacy.kms.chapter02.Money;
import legacy.kms.chapter02.Screening;

public class PercentDiscountPolicy extends DefaultDiscountPolicy {

    private double percent;

    public PercentDiscountPolicy(double percent, DiscountCondition ... conditions) {
        super(conditions);
        this.percent = percent;
    }

    @Override
    protected Money getDiscountAmount(Screening screening) {
        return screening.getMoveFee().times(percent);
    }
}

File: src/main/kotlin/hoondong/chapter02/PercentDiscountPolicy.kt
package hoondong.chapter02

class PercentDiscountPolicy (
    private val percent: Double,
    vararg conditions: DiscountCondition
): DefaultDiscountPolicy(*conditions) {
    override fun getDiscountAmount(screening: Screening): Money {
        return screening.getMovieFee().times(percent)
    }
}